### PR TITLE
Fix/4226 dark mode glitches

### PR DESCRIPTION
**Pull request by puneetmahali1**
## Description
This PR fix the all dark mode glitches.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4226#anchor

## Screenshots
IMG_0020
IMG_0021
IMG_0022
IMG_0023
IMG_0024
IMG_0025
IMG_0026
IMG_0027

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift [hunk 1]
```diff
@@ -75,7 +75,8 @@ class ExposureSubmissionTestResultConsentViewModel {
 									)
 								}
 							},
-						.space(height: 20)
+						.space(height: 20,
+							   color: .clear)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift:79]: Can't we set this to `.clear` by default?

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift (hunk 1 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift]: Thanks Nick. Yes we can & fixed it.

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewCell.swift [hunk 2]
```diff
@@ -133,7 +133,7 @@ extension DynamicCell {
 		.identifier(CellReuseIdentifier.space) { _, cell, _ in
 			guard let cell = cell as? DynamicTableViewSpaceCell else { return }
 			cell.height = height
-			cell.backgroundColor = color
+			cell.backgroundColor = .clear
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewCell.swift:136]: Would be a better idea to leave it as it was and change the function to `static func space(height: CGFloat, color: UIColor = .clear) -> Self {`

Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewCell.swift]: Thanx! Done.

**Comment by KaiTeuber** [comment]
Is a clear background color really the solution we want here?
Transparency is slower in rendering and we have had side effects with it in the past. e.g. the red tile once shone through in another section.
Wouldn't it be more effective to set the expected color directly?

**Comment by puneetmahali1** [comment]
> Is a clear background color really the solution we want here?
> Transparency is slower in rendering and we have had side effects with it in the past. e.g. the red tile once shone through in another section.
> Wouldn't it be more effective to set the expected color directly?

I was not aware of the side effects and didn't see any problems there. Yes, we can set the expected colors then we need those colors. Also, there is no figma screens for the dark mode. So if this is the case then need to ask from the designers and set the expected colors.

**Review by nickguendling — APPROVED** [review]

**Review by ArturFriesen — APPROVED** [review]

**Review by tcfos — APPROVED** [review]

**Pull request merged**